[Jolyn Leow] iP

jolynleow2301 opened the pull request:
### **Friday Chatbot**

Friday frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/nus-cs2103-AY2425S1/ip/pull/607).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is FREE!

Features:

- [x]  Managing tasks
- [x]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

jolynleow2301 committed:
Level-0: Rename, Greet, Exit

jolynleow2301 committed:
Updated

jolynleow2301 committed:
Level-1: Echo

jolynleow2301 committed:
Level-2: Add, List

jolynleow2301 committed:
Level-3: Mark as Done

jolynleow2301 committed:
Level-4: ToDos, Events, Deadlines

jolynleow2301 committed:
A-TextUiTesting: Automated Text UI Testing

jolynleow2301 committed:
Level-5: Handle Errors

jolynleow2301 committed:
Level-6: Delete

jolynleow2301 committed:
Level-6: Delete

jolynleow2301 committed:
Save tasks in the hard drive automatically

jolynleow2301 committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Save tasks in the hard drive automatically

casperngeen commented on src/main/java/fridayproject/Friday.java:
```diff
@@ -0,0 +1,200 @@
+                    tasks.add(todo);
+                    saveTasksToFile(tasks);
+                    System.out.println("Got it. I've added this task:\n  " + todo.getTypeIcon() 
+                    + todo.toString() + "\nNow you have " + tasks.size() + " tasks in the list.");
```
Wrapped lines should have an extra indent of 4 spaces

casperngeen commented on src/main/java/System/out/println.java:
```diff
@@ -0,0 +1,5 @@
+package System.out;
+
+public class println {
```
Is this an accident?

casperngeen reviewed (commented):
LGTM :smile:

You can consider creating a parser function or class.

shotnothing commented on src/main/java/fridayproject/Event.java:
```diff
@@ -0,0 +1,28 @@
+        this.end = end;
+    }
+
+    @Override
+    public String getTypeIcon() {
+        return "[E]";
+    }
+
+    @Override
+    public String toString() {
+        return super.toString() + " (from: " + start + " to: " + end + ")";
+    }
+
+    @Override 
+    public String toFileString() {
+        return "E | " + (this.isDone ? "1" : "0") + " | " + this.description;
+    }
```
i think it might be good to add javadoc here!

shotnothing commented on src/main/java/fridayproject/Friday.java:
```diff
@@ -0,0 +1,200 @@
+                case "T":
+                    if (parts.length >= 3) {
+                        Tasks todo = new Todo(parts[2]);
+                        if(parts[1].equals("1")) {
+                            todo.markAsDone();
+                        }
+                        tasks.add(todo);
+                    } else {
+                        System.out.println("Error loading task: " + taskString);
+                    }
+                    break;
+                case "D":
+                    if (parts.length >= 4) {
+                        Tasks deadline = new Deadline(parts[2], parts[3]);
+                        if (parts[1].equals("1")) {
+                            deadline.markAsDone();
+                        }
+                        tasks.add(deadline);
+                    } else {
+                        System.out.println("Error loading task: " + taskString);    
+                    }
+                    break;
+                case "E":
+                    if (parts.length >= 5) {
+                        Tasks event = new Event(parts[2], parts[3], parts[4]);
+                        if (parts[1].equals("1")) {
+                            event.markAsDone();
+                        }
+                        tasks.add(event);
+                    } else {
+                        System.out.println("Error loading task: " + taskString);
+                    }
+                    break;
+                default:
+                    System.out.println("Error loading task: " + taskString);
+                    break;
+                }
+            } catch (Exception e) {
+                System.out.println("Error loading task: " + taskString);
+            }
```
this looks like something that can be abstracted further, in the name of OOP and modularity

shotnothing commented on src/main/java/fridayproject/Tasks.java:
```diff
@@ -0,0 +1,32 @@
+    protected String description;
+    protected boolean isDone;
+
+    public Tasks(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    public String getStatusIcon() {
+        return (isDone ? "X" : " ");
+    }
+
+    public void markAsDone() {
+        this.isDone = true;
+    }
+
+    public void markAsUndone() {
+        this.isDone = false;
+    }
+
+    public abstract String getTypeIcon();
+
+    public abstract String toFileString();
+
+    @Override
+    public String toString() {
+        return "[" + this.getStatusIcon() + "] " + this.description;
+    }
```
these should have javadoc too i think!

Jasperabez commented on src/main/java/fridayproject/Tasks.java:
```diff
@@ -0,0 +1,32 @@
+package fridayproject;
+
+public abstract class Tasks {
```
Should `Tasks` be `Task` instead since it is not a collection?

Jasperabez commented on src/main/java/fridayproject/ArrayList.java:
Is this a redundant class?

Jasperabez commented on src/main/java/fridayproject/Friday.java:
```diff
@@ -0,0 +1,200 @@
+                    Tasks deadline = new Deadline(deadlineParts[0] , deadlineParts[1]);
+                    tasks.add(deadline);
+                    saveTasksToFile(tasks);
+                    System.out.println("Got it. I've added this task:\n  " + deadline.getTypeIcon() 
+                    + deadline.toString() + "\nNow you have " + tasks.size() + " tasks in the list.");
```
Similar issue to lines 130-131? Lines wrapped should have an extra indent of 4 spaces based on the coding standard.

Jasperabez commented on src/main/java/fridayproject/Friday.java:
```diff
@@ -0,0 +1,200 @@
+                    Tasks event = new Event(eventParts[0], eventParts[1], eventParts[2]);
+                    tasks.add(event);
+                    saveTasksToFile(tasks);
+                    System.out.println("Got it. I've added this task:\n  " + event.getTypeIcon() 
+                    + event.toString() + "\nNow you have " + tasks.size() + " tasks in the list.");
```
As per mentioned in 142-143

Jasperabez reviewed (commented):
LGTM besides some possible issues with naming and indentation

jolynleow2301 committed:
Change date format

jolynleow2301 committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Change date format

jolynleow2301 committed:
Update changes

jolynleow2301 committed:
Update changes

jolynleow2301 committed:
Use more OOP

jolynleow2301 committed:
Merge branch 'branch-MoreOop'

* branch-MoreOop:
  Use more OOP
  Update changes

# Conflicts:
#	data/friday.txt

jolynleow2301 committed:
Merge branch 'add-gradle-support' of https://github.com/jolynleow2301/ip

jolynleow2301 committed:
Add gradle support

jolynleow2301 committed:
Add test cases for Deadline.java and TaskList.java

jolynleow2301 committed:
Add JavaDoc comments

jolynleow2301 committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Add JavaDoc comments

jolynleow2301 committed:
Modify code to adhere to coding standards

jolynleow2301 committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Modify code to adhere to coding standards

jolynleow2301 committed:
Level-9: Find

jolynleow2301 committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Level-9: Find

jolynleow2301 committed:
Use checkStyle to detect coding style violations

jolynleow2301 committed:
Merge branch 'branch-A-CheckStyle'

* branch-A-CheckStyle:
  Use checkStyle to detect coding style violations

jolynleow2301 committed:
Add GUI to the chatbot

jolynleow2301 committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Add GUI to the chatbot

jolynleow2301 committed:
Add GUI to chatbot

jolynleow2301 committed:
Create another JAR file

jolynleow2301 committed:
Add assertions to ensure data integrity in core classes

Assertions have been added to crucial points in several classes to
ensure the correctness of the program's logic and prevent unexpected
states. This includes:

- `Tasks` class: Asserted that task descriptions are not null or empty
  and that tasks are not already marked as done or undone.
- `Storage` class: Added assertions to verify the file path is valid
  and the tasks list is properly loaded.

These assertions will help to catch programming errors early, promoting data integrity.

jolynleow2301 committed:
Add assertions to ensure data integrity in core classes

jolynleow2301 committed:
Refactor codebase to improve readability and maintainability

Enhanced code quality across the project by implementing the following
changes:

- Refactored `Parser` and `Ui` classes to simplify logic and improve
  method naming consistency.
- Improved error handling by adding meaningful assertions and
  exceptions to prevent invalid states.
- Reformatted code for better readability, including consistent
  indentation, line wrapping, and spacing.

These changes collectively improve the overall maintainability of the codebase, making it easier to understand and modify
in the future.

jolynleow2301 committed:
Merge branch 'branch-A-Assertions'

* branch-A-Assertions:
  Add assertions to ensure data integrity in core classes
  Add assertions to ensure data integrity in core classes

jolynleow2301 committed:
Merge branch 'branch-A-CodeQuality'

* branch-A-CodeQuality:
  Refactor codebase to improve readability and maintainability

jolynleow2301 committed:
Add assertions to ensure data integrity in core classes

Assertions have been added to crucial points in several classes to
ensure the correctness of the program's logic and prevent unexpected
states. This includes:

- `Tasks` class: Asserted that task descriptions are not null or empty
and that tasks are not already marked as done or undone.
- `Storage` class: Added assertions to verify the file path is valid
and the tasks list is properly loaded.

These assertions will help to catch programming errors early, promoting data integrity.

jolynleow2301 committed:
Refactor codebase to improve readability and maintainability

Enhanced code quality across the project by implementing the following
changes:

- Refactored `Parser` and `Ui` classes to simplify logic and improve
method naming consistency.
- Improved error handling by adding meaningful assertions and
exceptions to prevent invalid states.
- Reformatted code for better readability, including consistent
indentation, line wrapping, and spacing.

These changes collectively improve the overall maintainability of the codebase, making it easier to understand and modify
in the future.

jolynleow2301 committed:
Merge pull request #2 from jolynleow2301/branch-A-Assertions

Add assertions to ensure data integrity in core classes

jolynleow2301 committed:
Merge branch 'master' into branch-A-CodeQuality

* master:
  Add assertions to ensure data integrity in core classes

jolynleow2301 committed:
Merge pull request #3 from jolynleow2301/branch-A-CodeQuality

Refactor codebase to improve readability and maintainability

jolynleow2301 committed:
Add a way to view tasks in the form of a schedule

jolynleow2301 committed:
Merge branch 'branch-B-ViewSchedules'

* branch-B-ViewSchedules:
  Add a way to view tasks in the form of a schedule

jolynleow2301 committed:
Give Friday a unique personality

jolynleow2301 committed:
Give Friday a unique personality

jolynleow2301 committed:
Give Friday a unique personality

jolynleow2301 committed:
Give Friday a unique personality

jolynleow2301 committed:
Add improvements to GUI

jolynleow2301 committed:
Add improvements to GUI

jolynleow2301 committed:
Add improvement to GUI

jolynleow2301 committed:
Add a representative screenshot of the product

jolynleow2301 committed:
Update README.md

jolynleow2301 committed:
Update README.md

jolynleow2301 committed:
Update README.md

jolynleow2301 committed:
Update README.md

jolynleow2301 committed:
Update README.md

jolynleow2301 committed:
Update README.md

jolynleow2301 committed:
Update README.md

jolynleow2301 committed:
Add improvements to Parser.java

jolynleow2301 committed:
Merge branch 'master' of https://github.com/jolynleow2301/ip

* 'master' of https://github.com/jolynleow2301/ip:
  Update README.md
  Update README.md
  Update README.md

jolynleow2301 committed:
Add improvements to Parser.java

jolynleow2301 committed:
Update README.md
